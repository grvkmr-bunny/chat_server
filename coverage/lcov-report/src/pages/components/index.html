<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/pages/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">23.73% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>14/59</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">16.67% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">21.43% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">22.81% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/57</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CreateUser.jsx"><a href="CreateUser.jsx.html">CreateUser.jsx</a></td>
	<td data-value="17.78" class="pic low"><div class="chart"><div class="cover-fill" style="width: 17%;"></div><div class="cover-empty" style="width:83%;"></div></div></td>
	<td data-value="17.78" class="pct low">17.78%</td>
	<td data-value="45" class="abs low">8/45</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="22" class="abs low">4/22</td>
	<td data-value="18.18" class="pct low">18.18%</td>
	<td data-value="44" class="abs low">8/44</td>
	</tr>

<tr>
	<td class="file low" data-value="TableComponent.jsx"><a href="TableComponent.jsx.html">TableComponent.jsx</a></td>
	<td data-value="42.86" class="pic low"><div class="chart"><div class="cover-fill" style="width: 42%;"></div><div class="cover-empty" style="width:58%;"></div></div></td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="38.46" class="pct low">38.46%</td>
	<td data-value="13" class="abs low">5/13</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Apr 08 2019 18:34:36 GMT+0530 (India Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
