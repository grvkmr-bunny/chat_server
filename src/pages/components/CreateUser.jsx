import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import Button from '@material-ui/core/Button';
import * as yup from 'yup';
import { SnackBarContextConsumer } from '../SnackBarProvider/SnackBarProvider';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import gql from "graphql-tag";
import { Mutation } from "react-apollo";

const ADD_USER = gql`
  mutation AddUser($name: String!, $email: String!) {
    addUser(name: $name, email: $email) {
      id
      name
      email
    }
  }
`;

class CreateUser extends React.Component {
  schema = yup.object().shape({
    name: yup.string().required('Name is a required field'),
    email: yup.string().required('Email Address is a required field').email('Email Address must be a valid email'),
  });

  state = {
    open: false,
    showData: false,
    name: '',
    email: '',
    showError: {
      name: false,
      email: false,
    },
    error: {
      name: '',
      email: '',
    },
    touched: {
      name: false,
      email: false,
    },
  };

  handleClickOpen = () => {
    this.setState({ open: true });
  };

  handleClose = () => {
    this.setState({ 
      open: false,
      name: '',
      email: '',
      touched: {
        name: false,
        email: false,
      },
      showError: {
        name: false,
        email: false,
      }
    });
  }  

  handleSubmit = (e, addUser, openSnackBar) => {
    localStorage.removeItem("loginUser");
    e.preventDefault();
    const { name, email } = this.state;
    addUser({ variables: { name, email }}).then((data) => {
      const { id, name, email } = data.data.addUser
      const value = [id, name, email]
      localStorage.setItem("loginUser", JSON.stringify(value));
    })
    .catch(err => {
      console.log("err", err)
    });
    openSnackBar('Data created successfully!', 'success');
    this.handleClose();
    this.setState({
      showData: true,
    });

  }  

  handleChange = field => (event) => {
    this.setState({
      [field]: event.target.value,
    }, this.validateForm);
  }

  validateForm = () => {
    const {
      name,
      email,
    } = this.state;
    this.schema
      .validate({
        name, email,
      }, { abortEarly: false })
      .then(() => {
        this.handleError(null);
      })
      .catch((err) => {
        this.handleError(err);
      });
  }

  handleError = (err) => {
    const focussederror = {};
    const boolError = {};
    if (err) {
      err.inner.forEach((element) => {
        focussederror[element.path] = element.message;
        boolError[element.path] = true;
      });
    }
    this.setState({
      error: focussederror,
      showError: boolError,
    });
  }

  isTouched = field => () => {
    const {
      touched,
    } = this.state;
    this.setState({
      touched: { ...touched, [field]: true },
    }, this.validateForm);
  }

  hasError = () => {
    const {
      error,
      touched,
    } = this.state;
    if (!Object.values(error).some(item => item)
    && Object.values(touched).some(item => item)
    ) {
      return false;
    }
    return true;
  }

  getError = (field) => {
    const {
      touched, error,
    } = this.state;
    if (!touched[field]) {
      return '';
    }
    return error[field];
  }

  render() {
    const { open, name, email, showData, showError } = this.state;
    return (
      <div>
        <Button align="center" onClick={this.handleClickOpen} variant="contained" color="primary">
          Start
        </Button>
        <Dialog
          open={open}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle id="form-dialog-title"> Enter your details:</DialogTitle>
          <DialogContent>
            <TextField
              required
              fullWidth
              id="outlined-name"
              label="Name"
              name="name"
              value={name}
              margin="normal"
              variant="outlined"
              error={showError.name}
              helperText={this.getError('name')}
              onChange={this.handleChange('name')}
              onBlur={this.isTouched('name')}
            />
            <TextField
              required
              fullWidth
              id="outlined-email"
              label="Email"
              name="email"
              value={email}
              margin="normal"
              variant="outlined"
              error={showError.email}
              helperText={this.getError('email')}
              onChange={this.handleChange('email')}
              onBlur={this.isTouched('email')}
            />
          </DialogContent>
          <DialogActions>
            <Button variant='contained' onClick={this.handleClose} color="primary">
              Cancel
            </Button>
              <Mutation mutation={ADD_USER}>
                {(addUser, {data}) => (
                  <SnackBarContextConsumer>
                    {({ openSnackBar }) => (
                      <Button variant='contained' 
                        onClick={(e) => {this.handleSubmit(e, addUser, openSnackBar)}}
                        color="primary"
                        disabled={this.hasError()}
                      >
                        Create
                      </Button>
                    )}
                  </SnackBarContextConsumer>
                )}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            
              </Mutation>
          </DialogActions>
        </Dialog>
        {showData ?  <Route><Redirect to="/users" /></Route> : ''}
      </div>
    );
  }
}

export default CreateUser;