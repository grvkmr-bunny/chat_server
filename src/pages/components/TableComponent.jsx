/* eslint-disable no-underscore-dangle */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';

const styles = theme => ({
  root: {
    marginTop: theme.spacing.unit,
    overflowX: 'auto',
    elevation: 5,
  },
  table: {
    minWidth: 600,
    fontFamily: theme.typography.fontFamily,
  },
  tableRow: {
    cursor: 'pointer',
    '&:nth-of-type(odd)': {
      backgroundColor: theme.palette.grey[100],
    },
    '&:hover': {
      backgroundColor: theme.palette.grey[200],
    },
  },
  tableCell: {
    flex: 1,
  },
  noClick: {
    cursor: 'initial',
  },
});

class TableComponent extends Component {

  componentDidMount() {
    this.props.subscribeToMore();
  }

  tableBody = (data) => {
    const { classes, onSelect } = this.props;
    let loginData = JSON.parse(localStorage.getItem("loginUser"));
    return data.getAllUser.map(dataItem => {
      if (dataItem.name !== loginData[1]) {
        return (
          <TableRow
            key={dataItem.id}
            className={classes.tableRow}
          >
            <TableCell align='center' onClick={() => onSelect(dataItem.id, dataItem.name)}>
              {dataItem.name}
            </TableCell>
          </TableRow>
        )
      }
      return '';
    });
  };

  render() {
    const {
      classes,
      data,
    } = this.props;
    const loginData = JSON.parse(localStorage.getItem("loginUser"));
    return (
      <>
        <div style={ { display: 'flex', flexDirection: 'column', alignItems: 'center' }}>
          <span style={{ color: 'black', fontSize: '25px', marginBottom: '50px' }}>
            <h2>Welcome {loginData[1]}</h2>
            <hr/>
          </span>
          <Paper className={classes.root}>
            <Table className={classes.table}>
              <TableBody>{this.tableBody(data)}</TableBody>
            </Table>
          </Paper>
      </div>
      </>
    );
  }
}
TableComponent.propTypes = {
  classes: PropTypes.objectOf.isRequired,
  data: PropTypes.arrayOf.isRequired,
  onSelect: PropTypes.func.isRequired,
  // onClick: PropTypes.func.isRequired,
  subscribeToMore: PropTypes.func.isRequired,
};
export default withStyles(styles)(TableComponent);