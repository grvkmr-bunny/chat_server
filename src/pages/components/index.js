export { default as CreateUser } from './CreateUser';
export { default as TableComponent } from './TableComponent';
export { default as AvailableUsers } from './AvailableUsers';
export { default as NoMatch } from './NoMatch';
export { default as Messanger } from './Messanger';
export { default as ChatDialog } from './ChatDialog';
