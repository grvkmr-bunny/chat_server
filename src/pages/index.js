export { HomePage } from './HomePage';
export * from './SnackBarProvider';
export * from './components';