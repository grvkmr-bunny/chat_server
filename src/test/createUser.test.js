import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { TextField, Button, Typography, TableBody } from '@material-ui/core';
import CreateUser from '../pages/components/CreateUser';
import { SnackBarProvider } from '../pages/SnackBarProvider';
import sinon from 'sinon';
import TableComponent from '../pages/components/TableComponent';


const spy = sinon.spy();
describe('CreateUser component unit test', () => {
  let props = {
		data: {
			getAllUser: [{id: 7, name: 'ajay', email: 'ajay@gmail.com'}],
		},
		onSelect: spy,
		loginData: ['7','ajay','ajay@gmail.com'],
	};
	// beforeEach(() => {
	// 	defaultProps = {
	// 		open: false,

	// 	}
	// })

	test('renders without an error', () => {
		const wrapper = shallow(<CreateUser />);
		expect(wrapper.exists()).toBe(true);
	})

  
	it('should render a <TextField />', () => {
		const wrapper = shallow(<CreateUser />);
    expect(wrapper.find(TextField)).toHaveLength(2);
	});
  
	it('should render a <Typography />', () => {
		const wrapper = shallow(<CreateUser />);
    expect(wrapper.find(Typography)).toHaveLength(0);
	});
  
	it('should render a <Button />', () => {
		const wrapper = shallow(<CreateUser />);
    expect(wrapper.find(Button)).toHaveLength(2);
	});
  
	it('render correctly  Tablecomponent', () => {  
		const wrapper = shallow(<TableComponent />);
		expect(wrapper.find(TableComponent)).toMatchSnapshot();
	});

  it('props validation', () => {
    const wrapper = shallow(<SnackBarProvider />)
		// expect(wrapper.props().children).exists();
		expect(wrapper).toMatchSnapshot();
	});

	it('Tablecomponent props validation', () => {  
		const wrapper = mount(<TableComponent><TableBody {...props}></TableBody></TableComponent>);
		expect(wrapper.find("span")).toHaveLength(1);
	});
	
	it('Tablecomponent function validation', () => { 
		const wrapper = shallow(<TableComponent {...props}/>);
		// wrapper.setProps({
		// 	data: [{}],
		// 	onSelect: spy,
		// });
		// const event = {
		// 	target: { onClick: spy }
		// }
		wrapper.find("TableCell").at(0).simulate("click");
		expect(spy.called).toBe(true);
	});
})
