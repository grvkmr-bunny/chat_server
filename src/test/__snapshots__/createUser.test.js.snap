// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser component unit test props validation 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SnackBarProvider) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Object {},
      "classes": Object {
        "error": "SnackBarProvider-error-7",
        "icon": "SnackBarProvider-icon-8",
        "iconVariant": "SnackBarProvider-iconVariant-9",
        "message": "SnackBarProvider-message-10",
        "success": "SnackBarProvider-success-6",
      },
    },
    "ref": null,
    "rendered": Object {},
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Object {},
        "classes": Object {
          "error": "SnackBarProvider-error-7",
          "icon": "SnackBarProvider-icon-8",
          "iconVariant": "SnackBarProvider-iconVariant-9",
          "message": "SnackBarProvider-message-10",
          "success": "SnackBarProvider-success-6",
        },
      },
      "ref": null,
      "rendered": Object {},
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`CreateUser component unit test render correctly  Tablecomponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <WithStyles(TableComponent) />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "checkPropTypes": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "noClick": "TableComponent-noClick-5",
          "root": "TableComponent-root-1",
          "table": "TableComponent-table-2",
          "tableCell": "TableComponent-tableCell-4",
          "tableRow": "TableComponent-tableRow-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "classes": Object {
            "noClick": "TableComponent-noClick-5",
            "root": "TableComponent-root-1",
            "table": "TableComponent-table-2",
            "tableCell": "TableComponent-tableCell-4",
            "tableRow": "TableComponent-tableRow-3",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "legacyContextMode": "parent",
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getChildContext": Object {
              "calledByRenderer": false,
            },
            "getDerivedStateFromProps": Object {
              "hasShouldComponentUpdateBug": false,
            },
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
    Symbol(enzyme.__childContext__): null,
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "noClick": "TableComponent-noClick-5",
          "root": "TableComponent-root-1",
          "table": "TableComponent-table-2",
          "tableCell": "TableComponent-tableCell-4",
          "tableRow": "TableComponent-tableRow-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
}
`;
